impl/client/src/main/javascript/web/pentaho/lang/DomNode.js

```diff
@@ -0,0 +1,168 @@
+/*!
+ * Copyright 2018 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "module",
+  "pentaho/lang/Base",
+  "pentaho/util/object"
+], function(module, Base, O) {
+
+  var DomNode = Base.extend(module.id, /** @lends pentaho.lang.DomNode# */{
```

Comment:
What is this? Wasn't the Scene just an util?

Reply:
This is the `DomNode` class. `Scene` inherits from `DomNode`. Scene is optional and can be seen as "just an utility". Scenes are inherently hierarchical, so DOM-like functionality is part of its functionality. Defining a `DomNode` base class helps to keep DOM stuff separate from the `Scene`'s functionality which is purely visual related. Additionally, a `DomNode` class is useful in its own, and has a language-level character.